src/main/java/com/schibsted/spt/data/jslt/impl/BuiltinFunctions.java

```diff
@@ -760,6 +762,32 @@ public JsonNode call(JsonNode input, JsonNode[] arguments) {
     }
   }
 
+  // ===== IS-INTEGER
+
+  public static class IsInteger extends AbstractFunction {
+
+    public IsInteger() {
+      super("is-integer", 1, 1);
+    }
+
+    public JsonNode call(JsonNode input, JsonNode[] arguments) {
+      return NodeUtils.toJson(arguments[0].isInt());
+    }
+  }
+
+  // ===== IS-DECIMAL
+
+  public static class IsDecimal extends AbstractFunction {
+
+    public IsDecimal() {
+      super("is-decimal", 1, 1);
+    }
+
+    public JsonNode call(JsonNode input, JsonNode[] arguments) {
+      return NodeUtils.toJson(arguments[0].isDouble());
```

Comment:
Is it possible that we might get an instance of `FloatNode` and that this method will return `false` for it?

https://fasterxml.github.io/jackson-databind/javadoc/2.8/com/fasterxml/jackson/databind/node/FloatNode.html

Looks like the answer is yes: https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/node/FloatNode.java

Could we add a test where we manually construct the input as a `FloatNode` using the Jackson API, and then attempt to call `is-decimal(.)`? `StaticTests` is the place for this.

Reply:
yes, in fact. this wil fail for `FloatNode` and moreover `is-integer` will also fail for `BigDecimalNode`. I've changed this test to use `isIntegralNumber()` and `isFloatingPointNumber()`